src/main/java/io/vertx/core/net/impl/SslChannelProvider.java

```diff
@@ -143,25 +144,31 @@ public SslHandler createClientSslHandler(SocketAddress remoteAddress, String ser
     return sslHandler;
   }
 
-  public ChannelHandler createServerHandler() {
+  public ChannelHandler createServerHandler(java.net.SocketAddress remoteAddress) {
```

Comment:
rather use Vert.x `HostAndPort` type rather than a socket address

Reply:
@vietj Thanks for the comment!
The `HostAndPort` seems currently is used to parse/set host/authority in HTTP header, which looks like a totally different use case...
Here we want to pass the socket remote address info to the `SSLEngine` for `X509ExtendedKeyManager` to use.
Are you sure `HostAndPort` is preferred here?
If use `HostAndPort`, I may need to add a conversion method like `@Nullable static HostAndPort socketHostAndPort(SocketAddress address);` to simplify the conversion, is it OK to put it in `HostAndPort` interface (which will become a new external API)?

Reply:
it seems in this example we are using SocketAddress only to carry and host and port info, aren't we ?